File: Piko/Sources/SFNode.m
//
//  SFNode.m
//
//  Copyright (c) 2014 Satoshi Muraki. All rights reserved.
//

#import "SFNode.h"

#import "SFDebug.h"

@implementation SFNode

- (instancetype)initWithInputs:(NSArray *)inputs outputs:(NSArray *)outputs
{
    self = [super init];
    if (self != nil) {
        SFAssert(inputs != nil);
        SFAssert(outputs != nil);
        _inputs = [inputs copy];
        _outputs = [outputs copy];
        for (SFInput *input in inputs) {
            input.owner = self;
        }
        for (SFOutput *output in outputs) {
            output.owner = self;
        }
    }
    return self;
}

- (BOOL)rendersConstantFrame
{
    return NO;
}

- (BOOL)renderForPlayer:(SFPlaybackManager *)player
                  index:(UInt64)index
                  count:(NSInteger)count
                     to:(double *)buffer
{
    return NO;
}

@end

File: Piko/Sources/SFNode.swift
//
//  SFNode.swift
//
//  Copyright (c) 2014 Satoshi Muraki. All rights reserved.
//

import Foundation

@objc class SFNode: NSObject {

    var inputs: [SFInput]
    var outputs: [SFOutput]

    init(inputs: [SFInput], outputs: [SFOutput]) {
        self.inputs = inputs
        self.outputs = outputs
        super.init()
        for input in inputs {
            input.owner = self
        }
        for output in outputs {
            output.owner = self
        }
    }

    func rendersConstantFrame() -> Bool {
        return false
    }

    func renderForPlayer(player: SFPlaybackManager,
                          index: UInt64,
                          count: Int,
                      to buffer: UnsafeMutablePointer<Double>) -> Bool {
        return false
    }
}
